orchagent/aclorch.cpp

```diff
@@ -650,6 +631,262 @@ void AclRule::updateInPorts()
     }
 }
 
+
+bool AclRule::update(const AclRule& updatedRule)
```

Comment:
Could you confirm if the update of priority/match rules/actions is supported? I remembered that we only support updating binding ports before.

Reply:
Yes, with this API priority, match and actions of ACL rule can be updated via SAI set API.
We support updating bind ports on ACL table.

Reply:
@bingwang-ms @prsunny I did not find an API in SAI to query whether updating attributes is supported.

Reply:
That my only concern for this change since I noticed below lines
https://github.com/Azure/sonic-swss/blob/8727ae5c323d99e53dddcf84102372abced94568/orchagent/aclorch.cpp#L3292-3294
```
        default:
            SWSS_LOG_ERROR("Acl rule update not supported for attr name %s", attr_name.c_str());
```

Reply:
Lines that you are refering to are related to the current approach of doing AclRule update. As you can see in the code it is implemented only for SAI_ACL_ENTRY_ATTR_FIELD_IN_PORTS (That's why there is an error message that you pasted in the comment, not because it is not supported by SAI or SAI vendor). Let's say there is an Orch which wants to use some actions, matches that AclOrch does not know (like PbhOrch for example uses SET_LAG_HASH/SET_ECMP_HASH). Then, every time we add some new action or match that is only relevant to that new Orch we would have to change AclOrch and add a special logic for handling update for that new action and match. Appearantly the AclOrch becomes too huge a special pieces of which are related to other orch's functionality. So, instead of current approach we propose a generic enough AclOrch::updateAclRule that can work with any SAI match, any SAI action. I haven't removed the current ```bool AclOrch::updateAclRule(string table_id, string rule_id, string attr_name, void *data, bool oper)``` method because it is in use by MuxOrch, PfcWdOrch and moving them to a new API ```bool AclOrch::updateAclRule(shared_ptr<AclRule> updatedRule)``` requires a bit of refactor which I plan to do in a different PR. Eventually I propose to remove the old ```updateAclRule``` with the new one.